### PR TITLE
[BC-17150] As a template contributor, when I add a new methodology template, I should have tests to verify that my templates are against a correct methodology and step

### PR DESCRIPTION
**Pull request by rajkumar-u**
### User story
As a template contributor
When i add a new methodology template
I should have tests to verify that my templates are against a correct methodology and step
So that i can add template without any error

Jira ticket : [BC-17150](https://bugcrowd.atlassian.net/browse/BC-17150) 

### Technical explanation
- Added a test using JSON Schema, that the mapping is valid according to the schema
- Added a test for all templates referred to in the mapping file actually exist
- It will give an error message if template mapping is incorrect or not exist

### To test
1. Run the below commands in your console
```
pip3 install -r lib/requirements.txt
python3 -B lib/validate_bmt.py 
```

2. You can see the valid results
Screenshot 2021-09-08 at 12 20 33 PM

3. To test negative scenarios
In `templates.json` modify the mapping data as invalid
```
{
  "metadata": {
    "default": null
  },
  "content": [
    {
      "methodology": "website_testing",
      "children": [
        {
          "key": "information",
          "attribute": "notes",
          "template": "testing"
        }
      ]
    }
  ]
}
```
4. Again run the script `python3 -B lib/validate_bmt.py `
5. You can see the error message for template missing
Screenshot 2021-09-08 at 12 22 39 PM
6. kindly check and confirm, when adding or removing new template mappings it should exist.

### CODE REVIEW

**Comment by fedetaglia** [comment]
hey @rajkumar-u this all make sense, just a couple of questions/suggestions

- have you considered splitting the templates mapping files per methodology? ( no schema changes required )
  ei. not `templates/template.json` but `templates/website_testing.json` etc ..
  
 - is it possible to split the template mapping tests to their own file?

**Comment by rajkumar-u** [comment]
> hey @rajkumar-u this all make sense, just a couple of questions/suggestions
> 
> * have you considered splitting the templates mapping files per methodology? ( no schema changes required )
>   ei. not `templates/template.json` but `templates/website_testing.json` etc ..
> * is it possible to split the template mapping tests to their own file?

@fedetaglia as per our discussion, i have added below items
1. added `template.json and template.schema.json` inside the mappings directory(removed templates folder)
2. added `title` in metadata
3. added testing file separately.

File: lib/tests/test_template_mapping.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    self.validate_mapping_data(mapping['content'])
+
+  def validate_mapping_data(self, mapping):
+      for methodologyData in mapping:
+          self.check_template_exist(methodologyData['methodology'], methodologyData['children'])
+
+  def check_template_exist(self, methodology, steps):
+      """
+      Check a template mapping path from templates.json file
+      and check methodology template is exist or not in template repository
```
Comment (arcwhite) [inline lib/tests/test_template_mapping.py:41]:
```suggestion
      and check methodology template exists (or not) in template repository
```

**Review by arcwhite — APPROVED** [review]

File: lib/tests/test_template_mapping.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        self.template_filename
+    )
+
+    mapping = utils.get_json(self.template_filename)
+
+    self.validate_mapping_data(mapping['content'])
+
+  def validate_mapping_data(self, mapping):
+      for methodologyData in mapping:
+          self.check_template_exist(methodologyData['methodology'], methodologyData['children'])
```
Comment (arcwhite) [inline lib/tests/test_template_mapping.py:36]:
```suggestion
          self.check_template_exists(methodologyData['methodology'], methodologyData['children'])
```

File: lib/tests/test_template_mapping.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+  def check_template_exist(self, methodology, steps):
```
Comment (arcwhite) [inline lib/tests/test_template_mapping.py:38]:
```suggestion
  def check_template_exists(self, methodology, steps):
```

**Pull request merged**